cmd/kubeadm/app/constants/constants.go

```diff
@@ -555,3 +552,36 @@ func GetDNSVersion(dnsType kubeadmapi.DNSAddOnType) string {
 func GetKubeletConfigMapName(k8sVersion *version.Version) string {
 	return fmt.Sprintf("%s%d.%d", KubeletBaseConfigurationConfigMapPrefix, k8sVersion.Major(), k8sVersion.Minor())
 }
+
+// GetCurrentKubernetesVersion returns current Kubernetes version supported by kubeadm
+func GetCurrentKubernetesVersion() *version.Version {
+	gitVersion := pkgversion.Get().GitVersion
+	if strings.Contains(gitVersion, "-") {
+		gitVersion = strings.Split(gitVersion, "-")[0]
+	}
+	// If bazel or hack/lib/version.sh does not generate the correct version number,
+	// use hardcoded CurrentKubernetesVersion
+	// TODO: Get rid of hardcoded version
+	if gitVersion == "" || gitVersion == "v0.0.0" {
+		return CurrentKubernetesVersion
+	}
+	curVersion, err := version.ParseSemantic(gitVersion)
+	if err != nil {
+		return CurrentKubernetesVersion
+	}
+	return curVersion
+}
+
+// GetMinimumControlPlaneVersion returns the minimum control plane version kubeadm can deploy
+func GetMinimumControlPlaneVersion() *version.Version {
+	currentVersion := GetCurrentKubernetesVersion()
+	if currentVersion.Minor() > 0 {
+		return currentVersion.WithMinor(currentVersion.Minor() - 1)
+	}
+	return currentVersion
+}
+
+// GetMinimumKubeletVersion returns the minimum version of kubelet which kubeadm supports
+func GetMinimumKubeletVersion() *version.Version {
```

Comment:
nit: I guess I don't understand the reason to have this function? I find it more confusing because if I saw `GetMinimumControlPlaneVersion()` and `GetMinimumKubeletVersion()` I would think they returned different things, or at least could. But this is not a big deal as I know they where original 2 separate vars anyhow.